File: cli/src/main/java/io/microconfig/cli/ssl/AllowAllTrustManager.java
package io.microconfig.cli.ssl;

import javax.net.ssl.X509TrustManager;
import java.security.cert.X509Certificate;

public class AllowAllTrustManager implements X509TrustManager {

    @Override
    public X509Certificate[] getAcceptedIssuers() {
        return new X509Certificate[0];
    }

    @Override
    public void checkClientTrusted(java.security.cert.X509Certificate[] certs,
                                   String authType) {
    }

    @Override
    public void checkServerTrusted(java.security.cert.X509Certificate[] certs,
                                   String authType) {
    }
}

File: common/src/main/kotlin/io/microconfig/server/common/ssl/AllowAllTrustManager.kt
package io.microconfig.server.common.ssl

import java.security.cert.X509Certificate
import javax.net.ssl.X509TrustManager

class AllowAllTrustManager : X509TrustManager {
    override fun getAcceptedIssuers(): Array<X509Certificate> {
        return emptyArray()
    }

    override fun checkClientTrusted(certs: Array<X509Certificate>, authType: String) {
    }

    override fun checkServerTrusted(certs: Array<X509Certificate>, authType: String) {
    }
}